### ISSUE TITLE
several defects found by coverity

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#1678]
### Tool Name and Version

coverity

### Code Report

```plain
** CID 1488360:  Error handling issues  (CHECKED_RETURN)


________________________________________________________________________________________________________
*** CID 1488360:  Error handling issues  (CHECKED_RETURN)
/src/sock.c: 98 in sock_accept_conn()
92           (errno == ENOSYS || errno == EINVAL || errno == EBADF) &&
93           ((accept4_broken = 1))))
94     #endif
95      {
96              laddr = sizeof(*conn->src);
97              if ((cfd = accept(l->rx.fd, (struct sockaddr*)addr, &laddr)) != -1) {
>>>     CID 1488360:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fd_set_nonblock(cfd)" without checking return value. It wraps a library function that may fail and return an error code.
98                      fd_set_nonblock(cfd);
99                      if (master)
100                                     fd_set_cloexec(cfd);
101                     }
102             }
103     

** CID 1488359:  Error handling issues  (CHECKED_RETURN)


________________________________________________________________________________________________________
*** CID 1488359:  Error handling issues  (CHECKED_RETURN)
/src/fd.c: 791 in init_pollers_per_thread()
785     
786             if (pipe(mypipe) < 0)
787                     return 0;
788     
789             poller_rd_pipe = mypipe[0];
790             poller_wr_pipe[tid] = mypipe[1];
>>>     CID 1488359:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fd_set_nonblock(poller_rd_pipe)" without checking return value. It wraps a library function that may fail and return an error code.
791             fd_set_nonblock(poller_rd_pipe);
792             fd_insert(poller_rd_pipe, poller_pipe_io_handler, poller_pipe_io_handler, tid_bit);
793             fd_insert(poller_wr_pipe[tid], poller_pipe_io_handler, poller_pipe_io_handler, tid_bit);
794             fd_want_recv(poller_rd_pipe);
795             fd_stop_both(poller_wr_pipe[tid]);
796             return 1;

** CID 1488358:  Error handling issues  (CHECKED_RETURN)


________________________________________________________________________________________________________
*** CID 1488358:  Error handling issues  (CHECKED_RETURN)
/src/dns.c: 73 in dns_connect_nameserver()
67                       ns->counters->id, ns->id);
68              close(fd);
69              return -1;
70      }
71     
72      /* Make the socket non blocking */
>>>     CID 1488358:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fd_set_nonblock(fd)" without checking return value. It wraps a library function that may fail and return an error code.
73      fd_set_nonblock(fd);
74     
75      /* Add the fd in the fd list and update its parameters */
76      dgram->t.sock.fd = fd;
77      fd_insert(fd, dgram, dgram_fd_handler, MAX_THREADS_MASK);
78      fd_want_recv(fd);

** CID 1488357:  Error handling issues  (CHECKED_RETURN)


________________________________________________________________________________________________________
*** CID 1488357:  Error handling issues  (CHECKED_RETURN)
/src/sock.c: 100 in sock_accept_conn()
94     #endif
95      {
96              laddr = sizeof(*conn->src);
97              if ((cfd = accept(l->rx.fd, (struct sockaddr*)addr, &laddr)) != -1) {
98                      fd_set_nonblock(cfd);
99                      if (master)
>>>     CID 1488357:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fd_set_cloexec(cfd)" without checking return value. It wraps a library function that may fail and return an error code.
100                                     fd_set_cloexec(cfd);
101                     }
102             }
103     
104             if (likely(cfd != -1)) {
105                     if (unlikely(cfd >= global.maxsock)) {

** CID 1488356:  Error handling issues  (CHECKED_RETURN)


________________________________________________________________________________________________________
*** CID 1488356:  Error handling issues  (CHECKED_RETURN)
/src/log.c: 1670 in __do_send_log()
1664                            return;
1665                    } else {
1666                            /* we don't want to receive anything on this socket */
1667                            setsockopt(*plogfd, SOL_SOCKET, SO_RCVBUF, &zero, sizeof(zero));
1668                            /* does nothing under Linux, maybe needed for others */
1669                            shutdown(*plogfd, SHUT_RD);
>>>     CID 1488356:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fd_set_cloexec(*plogfd)" without checking return value. It wraps a library function that may fail and return an error code.
1670                            fd_set_cloexec(*plogfd);
1671                    }
1672            }
1673     
1674            msg_header = build_log_header(logsrv->format, level, facility, metadata, &nbelem);
1675      send:

** CID 1488355:  Error handling issues  (CHECKED_RETURN)


________________________________________________________________________________________________________
*** CID 1488355:  Error handling issues  (CHECKED_RETURN)
/src/proto_sockpair.c: 471 in sockpair_accept_conn()
465             struct proxy *p = l->bind_conf->frontend;
466             struct connection *conn = NULL;
467             int ret;
468             int cfd;
469     
470             if ((cfd = recv_fd_uxst(l->rx.fd)) != -1)
>>>     CID 1488355:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fd_set_nonblock(cfd)" without checking return value. It wraps a library function that may fail and return an error code.
471                     fd_set_nonblock(cfd);
472     
473             if (likely(cfd != -1)) {
474                     /* Perfect, the connection was accepted */
475                     conn = conn_new(&l->obj_type);
476                     if (!conn)

** CID 1488354:  Error handling issues  (CHECKED_RETURN)


________________________________________________________________________________________________________
*** CID 1488354:  Error handling issues  (CHECKED_RETURN)
/src/mworker.c: 424 in mworker_pipe_register_per_thread()
418             if (!(global.mode & MODE_MWORKER) || master)
419                     return 1;
420     
421             if (tid != 0)
422                     return 1;
423     
>>>     CID 1488354:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "fd_set_nonblock(proc_self->ipc_fd[1])" without checking return value. It wraps a library function that may fail and return an error code.
424             fd_set_nonblock(proc_self->ipc_fd[1]);
425             /* In multi-tread, we need only one thread to process
426              * events on the pipe with master
427              */
428             fd_insert(proc_self->ipc_fd[1], fdtab[proc_self->ipc_fd[1]].owner, mworker_accept_wrapper, tid_bit);
429             fd_want_recv(proc_self->ipc_fd[1]);
```

### Output of `haproxy -vv`

```plain
no
```

**Comment by wtarreau** [comment]
Thanks Ilya. Should be addressed with commit a80e4a354 ("MINOR: fd: add functions to set O_NONBLOCK and FD_CLOEXEC").

**Comment by chipitsine** [comment]
I'll check tomorrow (on the next scheduled scan). thanks!
